beacon-chain/rpc/validator/status.go

```diff
@@ -93,6 +93,11 @@ func (vs *Server) validatorStatus(ctx context.Context, pubKey []byte, headState
 		resp.ActivationEpoch = int64(val.ActivationEpoch)
 	}
 
+	// If the status can be retrieved from the beacon state, don't continue further.
+	if resp.Status != ethpb.ValidatorStatus_UNKNOWN_STATUS {
```

Comment:
this misses pending validators, you will end up not calculating their activation epoch below.